### PR TITLE
test: Set default pytest service key

### PR DESCRIPTION
**Pull request by daryllimyt**
## Summary
- Set a deterministic pytest default for `TRACECAT__SERVICE_KEY` before importing Tracecat config.
- Preserve caller-provided and repo `.env` service keys before falling back to `test-service-key`.
- Mirror the resolved value into `tracecat.config` during `env_sandbox` setup.

## Testing
- `uv run ruff check tests/conftest.py`
- `uv run ruff format --check tests/conftest.py`
- `uv run basedpyright tests/conftest.py`
- `env -u TRACECAT__SERVICE_KEY uv run pytest --collect-only tests/unit/test_auth.py -q`
- `env -u TRACECAT__SERVICE_KEY uv run pytest --collect-only tests/integration -q`

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -8,10 +8,10 @@
 from typing import Any
 from unittest.mock import patch
 
-# Set workflow return strategy BEFORE importing tracecat modules
-# test_workflows.py was written when we returned the full context by default
-# This must happen before any tracecat imports to ensure config reads the correct value
+# Set test defaults BEFORE importing tracecat modules so config reads them.
+# test_workflows.py was written when we returned the full context by default.
 os.environ.setdefault("TRACECAT__WORKFLOW_RETURN_STRATEGY", "context")
+os.environ.setdefault("TRACECAT__SERVICE_KEY", "test-service-key")
```
[the remark this answers is not in the document]
Comment (daryllimyt) [reply tests/conftest.py]: Thanks, this was a valid edge case. The PR branch now preserves precedence by reading `TRACECAT__SERVICE_KEY` from the shell env first, then the repo `.env`, and only falling back to `test-service-key` when neither is set. `env_sandbox` then mirrors that resolved value into `tracecat.config`, so local/in-Docker runs backed by `.env` should continue matching live services.
